src/plugins/console/public/lib/utils/index.ts

```diff
@@ -117,27 +117,25 @@ export const replaceVariables = (
   requests: RequestArgs['requests'],
   variables: DevToolsVariable[]
 ) => {
-  const urlRegex = /(\${\w+})/g;
-  const bodyRegex = /("\${\w+}")/g;
+  const urlRegex = /\${(\w+)}/g;
+  const bodyRegexSingleQuote = /([\\"]?)"\${(\w+)}"(?!")/g;
+  const bodyRegexTripleQuotes = /([\\"]?)"""\${(\w+)}"""(?!")/g;
+
   return requests.map((req) => {
     if (urlRegex.test(req.url)) {
-      req.url = req.url.replaceAll(urlRegex, (match) => {
-        // Sanitize variable name
-        const key = match.replace('${', '').replace('}', '');
+      req.url = req.url.replaceAll(urlRegex, (match, key) => {
         const variable = variables.find(({ name }) => name === key);
 
         return variable?.value ?? match;
       });
     }
 
     if (req.data && req.data.length) {
-      if (bodyRegex.test(req.data[0])) {
-        const data = req.data[0].replaceAll(bodyRegex, (match) => {
-          // Sanitize variable name
-          const key = match.replace('"${', '').replace('}"', '');
+      if (bodyRegexSingleQuote.test(req.data[0])) {
+        const data = req.data[0].replaceAll(bodyRegexSingleQuote, (match, lookbehind, key) => {
```

Comment:
I think following your latest changes removing the lookbehind, you don't need it in this code either.

Reply:
I removed only the use of lookbehind in the regex pattern. I tried not to degrade the capability of this replaceVariables function, thus I added `lookbehind` here.

The `lookbehind` will be either `'\\'`, `'"'` or `''` and the only last case is wanted, as the comment explains.